### PR TITLE
Basic FUNCTIONAL XMA2 support

### PR DESCRIPTION
**Pull request by 0x0ade**
See https://github.com/FNA-XNA/FAudio/pull/247 for more information and the submodule update.

### CODE REVIEW

File: src/Audio/SoundEffect.cs [hunk 1]
```diff
@@ -197,18 +202,37 @@ int loopLength
...
 			/* Buffer format */
-			format = new FAudio.FAudioWaveFormatEx();
-			format.wFormatTag = wFormatTag;
-			format.nChannels = nChannels;
-			format.nSamplesPerSec = nSamplesPerSec;
-			format.nAvgBytesPerSec = nAvgBytesPerSec;
-			format.nBlockAlign = nBlockAlign;
-			format.wBitsPerSample = wBitsPerSample;
-			format.cbSize = 0; /* May be needed for ADPCM? */
+			if (extraData == null) {
```
Comment (flibitijibibo) [inline src/Audio/SoundEffect.cs:211]: Braces

File: src/Content/ContentReaders/SoundEffectReader.cs [hunk 2]
```diff
@@ -53,10 +53,38 @@ SoundEffect existingInstance
 			uint nAvgBytesPerSec = Swap(se, input.ReadUInt32());
 			ushort nBlockAlign = Swap(se, input.ReadUInt16());
 			ushort wBitsPerSample = Swap(se, input.ReadUInt16());
-			/* ushort cbSize =*/ input.ReadUInt16();
-
-			// Seek past the rest of this crap (cannot seek though!)
-			input.ReadBytes((int) (formatLength - 18));
+			ushort cbSize = Swap(se, input.ReadUInt16());
+
+			byte[] extra = null;
```
Comment (flibitijibibo) [inline src/Content/ContentReaders/SoundEffectReader.cs:58]: Do we just want to use a byte* and stackalloc to prevent GC activity?

**Review by flibitijibibo — CHANGES_REQUESTED** [review]
This actually looks really good! Just a style thing and a possible memory optimization.

File: src/Content/ContentReaders/SoundEffectReader.cs (hunk 2 above)
Comment (0x0ade) [reply src/Content/ContentReaders/SoundEffectReader.cs:58]: Would be nice, but it'd require memcpy in the SoundEffect constructor to copy from extraData to formatPtr + offs. Buffer.BlockCopy doesn't exist in old .NET runtimes and SDL_memcpy exists but is off limits as this is outside of SDL2_FNAPlatform.

Thinking about it a bit more after talking on it on Discord: maaaaybe only for FNA.Core / a compile time flag? Might not be worth the effort tho 😅

**Review by flibitijibibo — APPROVED** [review]
Yeah, the stackalloc's not worth it now that I think about it. This is good to go pending Andrew's review for FAudio.

**Pull request merged**